libminifi/include/utils/Id.h

```diff
@@ -107,9 +107,12 @@ class IdentifierBase {
 typedef uint8_t UUID_FIELD[16];
 
 class Identifier : public IdentifierBase<UUID_FIELD, std::string> {
+  static constexpr const char* UUID_FORMAT_STRING = "%02hhx%02hhx%02hhx%02hhx-%02hhx%02hhx-%02hhx%02hhx-%02hhx%02hhx-%02hhx%02hhx%02hhx%02hhx%02hhx%02hhx";
```

Comment:
Pre-C++17, this is only a declaration that requires a definition. After C++17, this is redefined to be a definition and the namespace-scope one is a deprecated but accepted declaration.
See the bottom is this page: https://en.cppreference.com/w/cpp/language/static

Put this in the .cpp file: `constexpr const char* Identifier::UUID_FORMAT_STRING;`.

Reply:
I believe it is not necessary since UUID_FORMAT_STRING is not odr-used

> A variable whose name appears as a potentially-evaluated expression is odr-used unless it is an
object that satisfies the requirements for appearing in a constant expression (5.19) and the lvalue-to-rvalue
conversion (4.1) is immediately applied.